### PR TITLE
ReadHistory integration and text output

### PR DESCRIPTION
**Pull request by Hzj-jie**
Issue, #2483 
Also I have added a text output.

### CODE REVIEW

File: plugins/evernote.koplugin/main.lua [hunk 1]
```diff
@@ -139,9 +145,32 @@ function EvernoteExporter:addToMainMenu(tab_item_table)
...
+                text = _("Purge history records"),
+                callback = function()
+                    self.config:purge()
+                    UIManager:show(InfoMessage:new{
+                        text = _("History records are purged.\nAll notes will be exported again next time.\nSuggest to remove existing KOReaderClipping.txt to avoid a duplication."),
+                    })
+                end
+            },
+            {
+                text = _("  - All notes will be exported again next time"),
```
Comment (houqp) [inline plugins/evernote.koplugin/main.lua]: This one is a little bit confusing to me. When is next time exactly?

**Comment by houqp** [comment]
Looks like your other PR introduced a conflict with this one ;)

File: plugins/evernote.koplugin/main.lua (hunk 1 above)
Comment (Hzj-jie) [inline plugins/evernote.koplugin/main.lua]: Pressing this menu item will purge the record, and once export menu is pressed, all the bookmarks will be exported again.

The reason of not merging purge with export is to ensure users have time to decide whether the old KOReaderClippings.txt should be removed.

I do think the menu items can be optimized. I cannot quite tell the benefit of the two switches to control the output destination. Why cannot we just use 'export to evernote' 'export to html' and 'export to txt'? We can decide whether exporting all documents or current document based on isDocless(), I.e. exporting all in file manager, and exporting current document in reader.

What do you think?

**Comment by Hzj-jie** [comment]
Yes, I will merge it soon.

File: plugins/evernote.koplugin/main.lua (hunk 1 above)
Comment (houqp) [inline plugins/evernote.koplugin/main.lua]: I see, so this hint is only for the purge menu item right? In this case isn't it redundant? The info message in that callback already has the same hint.

> I do think the menu items can be optimized. I cannot quite tell the benefit of the two switches to control the output destination. Why cannot we just use 'export to evernote' 'export to html' and 'export to txt'?

Yes, I agree :)

> We can decide whether exporting all documents or current document based on isDocless(), I.e. exporting all in file manager, and exporting current document in reader.

The kpvbooklet users will probably never use filemanager. We should disable export current book item using isDocless() in filemanager though.

Comment (Hzj-jie) [inline plugins/evernote.koplugin/main.lua]: Emmm, should we add a switch in reader?

Comment (Hzj-jie) [inline plugins/evernote.koplugin/main.lua]: For the hint in menu item, yes, it's for purge feature only. I will remove it.

Comment (houqp) [inline plugins/evernote.koplugin/main.lua]: Thanks! I think reader should be fine. As I mentioned earlier, kpvbooklet users will probably rely on this behavior.

**Comment by houqp** [comment]
Thanks!

**Comment by seacatwu** [comment]
found this cause the old statistic history lost.

**Comment by houqp** [comment]
@Hzj-jie , can you help take a look?

**Comment by seacatwu** [comment]
update
latest version: the old statistic history lost problem solved.

**Pull request merged**